### PR TITLE
dpdk: umark broken on armv7 and set archs and machine

### PR DESCRIPTION
**Pull request by ndowens**
Signed-off-by: Nathan Owens <ndowens04@gmail.com>

### CODE REVIEW

File: srcpkgs/dpdk/template [hunk 1]
```diff
@@ -53,7 +51,7 @@ dpdk-devel_package() {
 		vmove usr/lib/dpdk/pmds-${version}/*.so
 		vmove usr/lib/pkgconfig
 		vmove usr/share/dpdk/examples
-	}
+}
```
Comment (Johnnynator) [inline srcpkgs/dpdk/template]: Accident? The old indentation was correct.

File: srcpkgs/dpdk/template [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Template file for 'dpdk'
 pkgname=dpdk
 version=19.08
-revision=1
+revision=2
 build_style=meson
 configure_args="-Denable_kmods=false -Denable_docs=true -Db_lto=false"
```
Comment (Johnnynator) [inline srcpkgs/dpdk/template]: Pls add `-Dmachine=default`, dpdk currently builds with `-march=native` on non cross builds.

File: srcpkgs/dpdk/template (hunk 1 above)
Comment (ndowens) [reply srcpkgs/dpdk/template]: Probably :)

File: srcpkgs/dpdk/template (hunk 2 above)
Comment (ndowens) [reply srcpkgs/dpdk/template]: > Pls add `-Dmachine=default`, dpdk currently builds with `-march=native` on non cross builds.

Done

File: srcpkgs/dpdk/template [hunk 3]
```diff
@@ -1,11 +1,12 @@
 # Template file for 'dpdk'
 pkgname=dpdk
 version=19.08
-revision=1
+revision=2
 build_style=meson
```
Comment (q66) [inline srcpkgs/dpdk/template]: add like

```
# no musl - uses error.h, execinfo.h, argp.h
archs="x86_64 aarch64 ppc64le i686 armv[67]l"
```

File: srcpkgs/dpdk/template [hunk 4]
```diff
@@ -14,12 +15,19 @@ homepage="https://www.dpdk.org/"
...
 checksum=1ceff1a6f4f8d5f6f62c1682097249227ac5225ccd9638e0af09f5411c681038
 
-if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
-	broken="Usage of error.h, execinfo.h and argp.h"
-fi
-
-case "$XBPS_TARGET_MACHINE" in
-	armv*) broken="Cannot get define of '__ARM_NEON'";;
+case "${XBPS_TARGET_MACHINE}" in
+	*-musl)
```
Comment (q66) [inline srcpkgs/dpdk/template]: with `archs=` you can ditch this

File: srcpkgs/dpdk/template [hunk 5]
```diff
@@ -14,12 +15,19 @@ homepage="https://www.dpdk.org/"
...
+		broken="Usage of error.h, execinfo.h and argp.h"
+	;;
+	armv6l)
```
Comment (q66) [inline srcpkgs/dpdk/template]: add a `*` for each case just in case we fix things on musl in the future

File: srcpkgs/dpdk/template [hunk 6]
```diff
@@ -14,12 +15,19 @@ homepage="https://www.dpdk.org/"
...
+		configure_args+=" -Dmachine=armv6-a"
+	;;
+	armv7l)
+		configure_args+=" -Dmachine=armv7-a"
+	;;
+	aarch64)
+		configure_args+=" -Dmachine=armv8-a"
+	;;
```
Comment (q66) [inline srcpkgs/dpdk/template]: also add a new case

```
ppc64le*) configure_args+=" -Dmachine=power8`;;
```

you can also turn these case statements into single lines, as long as they fit in 80col

**Pull request merged**